src/transformers/tokenization_utils.py

```diff
@@ -494,10 +494,17 @@ def get_added_vocab(self) -> Dict[str, int]:
 
     def __len__(self):
         """
-        Size of the full vocabulary with the added tokens. Counts the `keys` and not the `values` because otherwise if
-        there is a hole in the vocab, we will add tokenizers at a wrong index.
+        Size of the full vocabulary with the added tokens.
         """
-        return len(set(self.get_vocab().keys()))
+        return self.total_vocab_size
+
+    def _update_total_vocab_size(self):
+        """
+        Update the size of the full vocabulary with the added tokens. Counts the `keys` and not the `values` because
+        otherwise if there is a hole in the vocab, we will add tokenizers at a wrong index. This operation is slow and
+        is only updated when adding tokens.
+        """
+        self.total_vocab_size = len(self.get_vocab())
```

Comment:
Given `_add_tokens` already keeps a running count of the number of tokens added, couldn't we just add these to the total rather than re-count each time?

Reply:
yes agreed! Pushed this change

Reply:
Nice :) I guess the only thing to consider is are there methods to remove tokens too? Do we expect people to manually edit the vocab mapping?

Reply:
hmm technically no, but if users play around with popping out tokens then it would not be updated. Maybe we keep  `__len__ ` dynamic to be re-counted each time then? it's not heavy in terms of run time

Reply:
Sounds good!